### PR TITLE
[No QA] Create callable worklow for createNewVersion

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
This change is part of a larger undertaking to clean up our GitHub Actions to make them faster, simpler, and more approachable.

### Fixed Issues
$ (partial) https://github.com/Expensify/Expensify/issues/195693

### Tests
1. Merge PR A with the checklist unlocked.
1. A staging deploy should occur and when it completes PR A should get a deploy comment and should be added to the checklist.
1. Merge PR B with the checklist locked.
1. PR B should get a comment saying it's not being immediately deployed.
1. Merge PR C with the checklist locked and with a CP Staging label.
1. PR C should be CP'd to staging. It should be added to the checklist and when the deploy completes it should get a deploy comment.
1. Close the checklist.
1. A production deploy should occur, and PRs A and C should receive a deploy comment.
1. PR B should be deployed to staging and receive a deploy comment.

### CODE REVIEW

**Comment by roryabraham** [comment]
We also might be able to simplify this callable workflow by using `secrets: inherit`:

> If the secrets are inherited using secrets: inherit, you can reference them even if they are not defined in the on key.

This sounds convenient, we just have to decide if it's safe practice. I actually think it will be safe for any callable workflow that we own and which comes from the same repo

**Comment by roryabraham** [comment]
Created an issue for the last actionlint failure: https://github.com/rhysd/actionlint/issues/174

File: .github/scripts/validateActionsAndWorkflows.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ if [[ -z "${CI}" && -z "$(command -v shellcheck)" ]]; then
   brew install shellcheck
 fi
 
-curl -s curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash | bash -s -- 1.6.13
+bash <(curl https://raw.githubusercontent.com/rhysd/actionlint/main/scripts/download-actionlint.bash)
```
Comment (AndrewGable) [inline .github/scripts/validateActionsAndWorkflows.sh:37]: Confirmed that the previous fix is in this latest version 👍

**Review by AndrewGable — DISMISSED** [review]
Looks good! 👍

**Review by AndrewGable — APPROVED** [review]

**Comment by roryabraham** [comment]
https://github.com/Expensify/App/actions/runs/2659622994

**Pull request merged**